standalone/include/psen_scan_v2_standalone/protocol_layer/scan_round.h

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2020-2021 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef PSEN_SCAN_V2_STANDALONE_SCAN_ROUND_H
+#define PSEN_SCAN_V2_STANDALONE_SCAN_ROUND_H
+
+#include "psen_scan_v2_standalone/data_conversion_layer/monitoring_frame_msg.h"
+#include "psen_scan_v2_standalone/util/logging.h"
+
+namespace psen_scan_v2_standalone
+{
+namespace protocol_layer
+{
+/**
+ * @brief Buffers and validates monitoring frames for a scan round.
+ *
+ * Discovers if there are to many monitoring frames in a scan round.
+ * Informs when a scan round ended incomplete.
+ * Discovers and omits old messages.
+ */
+class ScanRound
+{
+public:
+  enum class Result
+  {
+    /* Message was not added due to being part of a previous scan round */
+    msg_was_too_old,
+    /* New scan round started, but the last one was not completed */
+    started_new_round_early,
+    /* Too many messages arrived in this scan round */
+    is_oversaturated,
+    /* The expected message count is reached */
+    is_complete,
+    /* The expected message count is not yet reached */
+    is_waiting_for_more_frames
+  };
+
+public:
+  ScanRound(const uint32_t& num_expected_msgs);
+  /**
+   * @brief Adds the message to the current scan round if it is considered valid. Returns the current status.
+   *
+   * @note:
+   * A scan round is considered to be complete whenever the next scan round starts or the expected number of messages
+   * arived.
+   *
+   * @see ScanRound::Result
+   *
+   * @param msg Current received MonitoringFrames.
+   * @return Status of current scan round
+   */
+  ScanRound::Result addValid(const data_conversion_layer::monitoring_frame::Message& msg);
+
+  /**
+   * @brief Readies the validator for a new validation round. This function has to be called whenever
+   * there is an expected brake in the receiving of MonitoringFrames.
+   */
+  void reset();
+  std::vector<data_conversion_layer::monitoring_frame::Message> getMsgs();
+
+private:
+  ScanRound::Result validate();
+
+private:
+  std::vector<data_conversion_layer::monitoring_frame::Message> curr_scan_round_{};
```

Comment:
`curr_scan_round_` indicates a member of type `ScanRound`, in fact here it is a member of `class ScanRound`. I suggest going with something `*buffer*`.